java/src/jmri/jmrit/beantable/BeanTableDataModel.java

```diff
@@ -586,6 +634,20 @@ protected JTable configureJTable(@Nonnull String name, @Nonnull JTable table, @N
      return "Bean";
      }*/
 
+    /**
+     * Updates the visibility settings of the property columns.
+     * @param table the JTable object for the current display.
+     * @param visible true to make the proeprty columns visible, false to hide.
+     */
+    public void setPropertyColumnsVisible(JTable table, boolean visible) {
+        XTableColumnModel columnModel = (XTableColumnModel) table.getColumnModel();
```

Comment:
Forcing all tables to use the XTableColumnModel is a bad idea. There is no need to cast to this, so don't.

Reply:
The column model is created about 15 lines up as a new XTableColumnModel(). There is already a cast like this in the showTableHeaderPopup function. Note that this only applies to BeanTables, not to all tables.
I need to have the XTableColumnModel, because the visiblity change functions are implemented there, which is the feature I need to call.